payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt

```diff
@@ -108,14 +139,18 @@ internal class DefaultCardNumberController constructor(
             ?: CardBrand.Unknown
     }
 
-    override val cardBrandFlow = impliedCardBrand.combine(
-        chosenBrand
-    ) { impliedBrand, chosenBrand ->
-        if (isEligibleForCardBrandChoice) {
-            chosenBrand
-        } else {
-            impliedBrand
+    override val cardBrandFlow = if (isEligibleForCardBrandChoice) {
+        combine(
+            brandChoices,
+            selectedCardBrandFlow
+        ) { choices, selected ->
+            when (choices.size) {
+                1 -> choices.first()
+                else -> selected
+            }
         }
+    } else {
+        impliedCardBrand
```

Comment:
It was my understanding that `impliedCardBrand` was to be used with CBC. Did that change?

Reply:
CBC is now relying on what the account service returns and sets in the private `brandChoices` flow. Should that work fine?

Reply:
Yep, that part’s fine. But I thought that `impliedCardBrand` refers to the brand that we use under the hood to determine PAN and CVC length when the user enters a co-branded card and doesn’t select a brand. Am I understanding this wrong?

Reply:
Per pairing, lots of merging of logic between account range service & card brand. Will look into this in the next couple of weeks.